tests/model/test_model.py

```diff
@@ -2079,20 +2098,73 @@ def test_toggle_stream_pinned_status(
                     "id": 0,
                 },
                 True,
+                False,
             ),
         ],
-        ids=["not_in_pm_narrow", "not_in_pm_narrow_with_sender", "start", "stop"],
+        ids=[
+            "not_in_pm_narrow",
+            "not_in_pm_narrow_with_sender",
+            "start",
+            "start when animating",
+            "stop",
+        ],
     )
-    def test__handle_typing_event(self, mocker, model, narrow, event, called):
+    def test__handle_typing_event(
+        self,
+        mocker,
+        model,
+        narrow,
+        event,
+        is_animation_in_progress,
+        is_recipient_actually_typing,
+    ):
         event["type"] = "typing"
 
-        mocker.patch("zulipterminal.ui.View.set_footer_text")
+        mocker.patch("zulipterminal.model.Model._show_typing_animation")
         model.narrow = narrow
         model.user_dict = {"hamlet@zulip.com": {"full_name": "hamlet"}}
+        model.is_recipient_typing = False
+        model.is_typing_animation_in_progress = is_animation_in_progress
 
         model._handle_typing_event(event)
 
-        assert model.controller.view.set_footer_text.called == called
+        assert model.is_recipient_typing == is_recipient_actually_typing
+        assert model._show_typing_animation.called == (
+            is_recipient_actually_typing and not is_animation_in_progress
```

Comment:
It'd be cleaner to parametrize over `_show_typing_animation_is_called` than derive it using these two variables?

Reply:
I think it would become too many variables to read and too many True's and False's. I've tried extracting the inside expression as another boolean variable(`expect_indicator`) inside the function and it looks better.

Reply:
The former shouldn't be an issue. I would avoid writing expressions/evaluations in the test code and focus on just testing the method with a bunch of inputs.